main/src/ca/uwaterloo/flix/language/errors/SafetyError.scala

```diff
@@ -8,24 +8,79 @@ import ca.uwaterloo.flix.util.vt.VirtualTerminal
 /**
   * A common super-type for safety errors.
   */
-sealed trait SafetyError extends CompilationError
+sealed trait SafetyError extends CompilationError {
+  def kind: String = "Safety Error"
+}
 
 object SafetyError {
 
+  /**
+    * Creates an IllegalNonPositivelyBoundVariable or IllegalNegativelyBoundWildVariable error
+    * depending on `sym.isWild`.
+    *
+    * @param loc the position of the body atom containing the illegal variable.
+    */
+  def makeIllegalNonPositivelyBoundVariableError(sym: Symbol.VarSym, loc: SourceLocation): SafetyError =
```

Comment:
We try to keep all the error classes "dumb" and I think this check is only used in once place, so I would move it there instead. Its better to keep the "business logic" in the phases.